.github/workflows/ci.yml

```diff
@@ -150,6 +150,13 @@ jobs:
               -DENABLE_icubmod_parametricCalibrator:BOOL=ON -DENABLE_icubmod_parametricCalibratorEth:BOOL=ON \
               -DBUILD_TESTING:BOOL=ON \
               -DCMAKE_INSTALL_PREFIX=${GITHUB_WORKSPACE}/install ..
+    
+    - name: Execute unit test
+      if: matrix.os == 'ubuntu-latest' || matrix.os == 'macOS-latest'
+      run: |
+        cd build
+        cmake --build . --config ${{ matrix.build_type }} --target test
```

Comment:
For running the unit tests, you first need to build the library, so I suggest to move this step after the "build" step. Furthermore, if you use CTest you can print just the output of failed tests (useful if you have a lot of tests) and be able to run the tests on Windows, so you can remove the if . See https://github.com/robotology/how-to-export-cpp-library/blob/master/.github/workflows/ci.yml#L41 : 
~~~
ctest --output-on-failure -C ${{ matrix.build_type }} .
~~~